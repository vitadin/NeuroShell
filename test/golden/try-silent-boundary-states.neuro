%% Testing try-silent boundary state management
%% This test verifies that boundary markers are properly processed
%% and that silent/try states don't interfere with each other

\set[_style=dark1]
\set[_echo_command="true"]

%% Show initial state  
\echo === Try-Silent Boundary States Test ===

%% Test 1: silent-try (reverse order)
\echo Testing silent-try combination...
\silent \try \echo This should be silent
\echo After silent-try (should display)

%% Test 2: try-silent-try (nested)
\echo Testing nested try-silent-try...
\try \silent \try \echo Nested silent in try
\echo After nested combination

%% Test 3: Error in nested silent-try
\echo Testing error in nested silent-try...
\silent \try \unknown-nested-error
\echo After nested error (should display)

%% Test 4: Complex nesting with variables
\echo Testing complex nesting with variables...
\set[before=initial]
\try \silent \set[during=modified]
\set[after=final]
\echo Before: ${before}, During: ${during}, After: ${after}

%% Final verification
\echo === Boundary states test completed ===
