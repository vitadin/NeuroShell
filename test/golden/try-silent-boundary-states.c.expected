Setting _style = dark1
Setting _echo_command = true
%%> "\\echo === Try-Silent Boundary States Test ==="
=== Try-Silent Boundary States Test ===
%%> "\\echo Testing silent-try combination..."
Testing silent-try combination...
%%> "\\silent \\try \\echo This should be silent"
%%> "\\echo After silent-try (should display)"
After silent-try (should display)
%%> "\\echo Testing nested try-silent-try..."
Testing nested try-silent-try...
%%> "\\try \\silent \\try \\echo Nested silent in try"
%%> "\\silent \\try \\echo Nested silent in try"
%%> "\\echo After nested combination"
After nested combination
%%> "\\echo Testing error in nested silent-try..."
Testing error in nested silent-try...
%%> "\\silent \\try \\unknown-nested-error"
%%> "\\echo After nested error (should display)"
After nested error (should display)
%%> "\\echo Testing complex nesting with variables..."
Testing complex nesting with variables...
%%> "\\set[before=initial]"
Setting before = initial
%%> "\\try \\silent \\set[during=modified]"
%%> "\\silent \\set[during=modified]"
%%> "\\set[after=final]"
Setting after = final
%%> "\\echo Before: ${before}, During: ${during}, After: ${after}"
Before: initial, During: modified, After: final
%%> "\\echo === Boundary states test completed ==="
=== Boundary states test completed ===